src/Host/Broker/Impl/Startup/Program.cs

```diff
@@ -57,7 +61,38 @@ public class Program {
 
             var tlsConfig = new TlsConfiguration(_logger, _securityOptions);
             var httpsOptions = tlsConfig.GetHttpsOptions(Configuration);
-            CreateWebHost(httpsOptions).Run();
+
+            Uri uri = GetServerUri(Configuration);
+            if (IsPortAvailable(uri?.Port)) {
```

Comment:
This check was to handle the case where the port is occupied even before broker starts. I will update with mitigation for the corner case.